src/test/scala/com/fulcrumgenomics/umi/DuplexConsensusCallingIntegrationTest.scala

```diff
@@ -0,0 +1,211 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.bam.api.{SamOrder, SamRecord}
+import com.fulcrumgenomics.testing.SamBuilder.{Minus, Plus, Strand}
+import com.fulcrumgenomics.testing.{ReferenceSetBuilder, SamBuilder, UnitSpec}
+import com.fulcrumgenomics.FgBioDef.{FilePath, PathToBam}
+import com.fulcrumgenomics.commons.util.CaptureSystemStreams
+
+/** Integration tests for calling duplex consensus reads.
+  *
+  * Runs:
+  * - GroupReadsByUmi
+  * - CallDuplexConsensusReads
+  * - FilterConsensusReads
+  * */
+class DuplexConsensusCallingIntegrationTest extends UnitSpec with CaptureSystemStreams {
+
+  /** Make a reference file that is 100 lines of 100 As. */
+  private val ref = {
+    val builder = new ReferenceSetBuilder
+    builder.add("chr1").add("AAAAAAAAAA", 10 * 100) // 5000 bases
+    builder.toTempFile()
+  }
+
+  /** Creates a read pair where the bases are all As and sets the RX tag to "AAAAAAAA"*/
+  private def record(builder: SamBuilder, name: String, start1: Int, start2: Int, strand1: Strand=Plus, strand2: Strand=Minus, base: String = "A", umiBases: String = ""): Seq[SamRecord] = {
+    val rxValue = if (umiBases.nonEmpty) umiBases else {
+      if (strand1 == Plus) "ACG-GCT" else "GCT-ACG"
+    }
+    require(base.length == 1)
+    builder.addPair(name=name, start1=start1, start2=start2, strand1=strand1, strand2=strand2, attrs=Map("RX" -> rxValue)).map { rec =>
+      rec.bases = base*builder.readLength
+      rec
+    }
+  }
+
+  /** Create some various test cases. */
+  private val (inputBam, inputRecords) = {
+    val builder = new SamBuilder(readLength=100, sort=Some(SamOrder.Coordinate))
+
+    // group 0: one FR read pair -> duplex from one strand (if pass filters)
+    {
+      record(builder, "group0:r1", start1=2, start2=101, strand1=Plus, strand2=Minus)
+    }
+
+    // group 1: one FR read pair and one RF read pair -> duplex from both strand (if pass filters)
+    {
+      record(builder, "group1:r1:FR", start1=3, start2=102, strand1=Plus, strand2=Minus)
+      record(builder, "group1:r1:RF", start1=102, start2=3, strand1=Minus, strand2=Plus)
+    }
+
+    // group 2: five FR read pairs and one RF read pair -> duplex from both strand (if pass filters)
+    {
+      Stream.range(start=1, end=6).foreach { i => record(builder, s"group3:r$i:FR", start1=4, start2=103, strand1=Plus, strand2=Minus) }
```

Comment:
I haven't seen `Stream.range` before.  Is there a reason to prefer it to `Range.inclusive(1,5)`?  I like that the `Range` version is clearer about inclusivity/exclusivity.